NameGame/NameGame/UI/UserGrid/ViewController/UserGridViewController.swift

```diff
@@ -0,0 +1,168 @@
+//
+//  UserGridViewController.swift
+//  NameGame
+//
+//  Created by Tammy Le on 6/14/17.
+//  Copyright © 2017 WillowTree Apps. All rights reserved.
+//
+
+import UIKit
+import ReSwift
+
+final class UserGridViewController: UICollectionViewController {
+    
+    static let reuseCellIdentifier = "userRecordCell"
+    
+    @IBOutlet weak var activityIndicator: UIActivityIndicatorView!
+    fileprivate var imageCache: ImageCache!
+    
+    fileprivate var viewModel = UserGridViewModel()
+    weak var lifecycleObserver: ViewLifecycleObserver? = nil
+    
+    class func make() -> UserGridViewController {
+        return UIStoryboard.main.make()
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        (self.collectionView?.collectionViewLayout as! UICollectionViewFlowLayout).sectionFootersPinToVisibleBounds = true
+        
+        imageCache.eventObserver = self
+        
+        lifecycleObserver?.viewDidLoad()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        lifecycleObserver?.viewWillAppear(animated)
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        lifecycleObserver?.viewDidAppear(animated)
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        lifecycleObserver?.viewWillDisappear(animated)
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        lifecycleObserver?.viewDidDisappear(animated)
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+
+    override func numberOfSections(in collectionView: UICollectionView) -> Int {
+        guard viewModel.userRecords != nil else {
+            return 0
+        }
+        return 1
+    }
+
+    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        guard let userRecords = viewModel.userRecords else {
+            return 0
+        }
+        return userRecords.count
+    }
+
+    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = UserRecordCell.dequeueCell(in: collectionView, with: UserGridViewController.reuseCellIdentifier, for: indexPath)
+        
+        let cachedImage = imageCache.image(for: viewModel.userRecords![indexPath.item])
```

Comment:
Let's use the nil coalescing operator so we don't need to force unwrap the user records.

```swift
let userRecords = viewModel.userRecords ?? []
...
userRecords[indexPath.item]
```